### ISSUE TITLE
Fatal error: Uncaught Error: Call to a member function validate_password() on null in login.php

### ISSUE THREAD

**Issue by stevenmybarra** [rothkj1022/phppickem#57]
I'm totally lost in how to fix this.  my database connects and the tables have been created. i see this error when i try to login as admin. i cannot create new accounts.

is there a surefire php config i can use that will work?

im currently on php version 7.4 with mysqli and mcrypt installed. do i need to go down?

any help is appreciated.

**Comment by searnhardt** [comment]
Have you compared the PHPINFO page with what you are expecting.  When I upgraded, PHP was still using an old version and I didn’t know it.  I’m not a linux guru and don’t want to assume about your level but google PHPINFO or look at it for your platform if you haven’t.

From: STEVEN YBARRA ***@***.***>
Sent: Monday, September 4, 2023 10:59 PM
To: rothkj1022/phppickem ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [rothkj1022/phppickem] Fatal error: Uncaught Error: Call to a member function validate_password() on null in login.php (Issue #57)


I'm totally lost in how to fix this. my database connects and the tables have been created. i see this error when i try to login as admin. i cannot create new accounts.

is there a surefire php config i can use that will work?

im currently on php version 7.4 with mysqli and mcrypt installed. do i need to go down?

any help is appreciated.

**Comment by SteveLeVant** [comment]
Did anyone figure this out. I'm having the same problem. I have run this script in the past but I lost the old version I had and downloaded this new one. Now I'm having this exact problem. 

Fatal error: Uncaught Error: Call to a member function validate_password() on null in /home/itkusnjw/public_html/xpickem/login.php:10 Stack trace: #0 {main} thrown in /home/itkusnjw/public_html/xpickem/login.php on line 10

-------------------------------------------------------------------------------------------
<?php
error_reporting(E_ALL ^ E_NOTICE);
//session_start();

require_once('includes/application_top.php');

$_SESSION = array();

if(is_array($_POST) && sizeof($_POST) > 0){
	$login->validate_password();  <----- Line 10
}
-----------------------------------------------------------------------

Any ideas or does someone want to share their login.php that works.

OR, is there a repository with this script from 12 years ago that worked for me.

Thanks!

**Comment by stevenmybarra** [comment]
> Did anyone figure this out. I'm having the same problem. I have run this script in the past but I lost the old version I had and downloaded this new one. Now I'm having this exact problem.
> 
> Fatal error: Uncaught Error: Call to a member function validate_password() on null in /home/itkusnjw/public_html/xpickem/login.php:10 Stack trace: #0 {main} thrown in /home/itkusnjw/public_html/xpickem/login.php on line 10
> 
> 0){ $login->validate_password(); <----- Line 10 } ----------------------------------------------------------------------- Any ideas or does someone want to share their login.php that works. OR, is there a repository with this script from 12 years ago that worked for me. Thanks!

I ended up using this
https://github.com/rbiddle/phppickem
php 7.2

there was some modifications as well that can be found in the issues tab of the link i just posted.

i know it's not ideal but it worked.

**Comment by rbiddle** [comment]
The comment above is my recommendation too. I did some upgrades for php 7.2 and a big part of that was dealing with the mcrypt function getting deprecated.

A few of us had been maintaining it (or variations of it) since then. We're in discord for ongoing maintenance stuff, but anything big, I commit back to my repo (there hasn't been much in a while).

If you want to join the discord, let me know and I'll send you the invite url. If you just want the fix, my repo should get it working for you.

**Comment by SteveLeVant** [comment]
Thanks for your response. Now something is not right on my end. Getting a 500 error. Nothing you can do for me unless you have encountered it yourself. PHP 7.2 and database setup correctly. Loads the favicon so I know it's connecting, but then 500 error. 

> The comment above is my recommendation too. I did some upgrades for php 7.2 and a big part of that was dealing with the mcrypt function getting deprecated.
> 
> A few of us had been maintaining it (or variations of it) since then. We're in discord for ongoing maintenance stuff, but anything big, I commit back to my repo (there hasn't been much in a while).
> 
> If you want to join the discord, let me know and I'll send you the invite url. If you just want the fix, my repo should get it working for you.

**Comment by SteveLeVant** [comment]
Yes, please send me the invite url for discord. Again, thanks for you attention to this issue. I'm just a left over Cobol dinosaur from the 90's trying to learn some of the new stuff all the cool kids are playing with. Thanks again.


> The comment above is my recommendation too. I did some upgrades for php 7.2 and a big part of that was dealing with the mcrypt function getting deprecated.
> 
> A few of us had been maintaining it (or variations of it) since then. We're in discord for ongoing maintenance stuff, but anything big, I commit back to my repo (there hasn't been much in a while).
> 
> If you want to join the discord, let me know and I'll send you the invite url. If you just want the fix, my repo should get it working for you.

**Comment by rbiddle** [comment]
Here's a link: https://discord.gg/tzQzqMug

We're not on all the time, but if you want help on the 500 error, someone on the discord is likely to help out.

**Comment by SteveLeVant** [comment]
Thank you, I figured out the 500 error. As usual nothing difficult, just
hiding in plain sight.

Appreciate all your help!
